### PR TITLE
Add support for SpaceAutomation

### PR DESCRIPTION
**Pull request by matkoch**
Let me know if anything is missing to support a new CI system.

### CODE REVIEW

**Comment by asbjornu** [comment]
Thanks. Some tests would be nice.

**Comment by matkoch** [comment]
Not exactly sure what you wanna test here. Do you want to add a sample CI?

**Comment by arturcic** [comment]
@matkoch have a look here https://github.com/GitTools/GitVersion/blob/1f8ff17dbab759b20ee361e7a6fc0488c985a803/src/GitVersionExe.Tests/PullRequestInBuildAgentTest.cs#L26 and https://github.com/GitTools/GitVersion/tree/1f8ff17dbab759b20ee361e7a6fc0488c985a803/src/GitVersionCore.Tests/BuildAgents

**Comment by matkoch** [comment]
There is no pull-request support yet. Honestly, other tests look more like repeating the implementation (or testing `BuildAgentBase`). Are you sure you don't want to have a sample CI instead?

**Comment by asbjornu** [comment]
Yes, the latest builds can be downloaded from [the GitHub Package registry](https://github.com/orgs/GitTools/packages?repo_name=GitVersion).

**Comment by asbjornu** [comment]
@matkoch, the tests ensure that the build server implementation output the expected stuff. We need tests for that. A sample CI would be fine as well, but does not alleviate the need for tests.

**Comment by matkoch** [comment]
Tests added. Sorry for the long delay!

**Comment by asbjornu** [comment]
Awesome, @matkoch! If you could please rebase against `HEAD` of `main`, this should be good to go! :shipit:

File: src/GitVersionCore.Tests/BuildAgents/SpaceAutomationTests.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        {
+            var sp = ConfigureServices(services =>
+            {
+                services.AddSingleton<SpaceAutomation>();
+            });
+            environment = sp.GetService<IEnvironment>();
+            buildServer = sp.GetService<SpaceAutomation>();
+            environment.SetEnvironmentVariable(SpaceAutomation.EnvironmentVariableName, "true");
+
+            githubSetEnvironmentTempFilePath = Path.GetTempFileName();
```
Comment (arturcic) [inline src/GitVersionCore.Tests/BuildAgents/SpaceAutomationTests.cs:29]: I wonder why would you need the `githubSetEnvironmentTempFilePath` here in the `SpaceAutomation` tests

File: src/GitVersionCore.Tests/BuildAgents/SpaceAutomationTests.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            environment.SetEnvironmentVariable(SpaceAutomation.EnvironmentVariableName, null);
+            if (githubSetEnvironmentTempFilePath != null && File.Exists(githubSetEnvironmentTempFilePath))
```
Comment (arturcic) [inline src/GitVersionCore.Tests/BuildAgents/SpaceAutomationTests.cs:36]: I wonder why would you need the `githubSetEnvironmentTempFilePath` here in the `SpaceAutomation` tests

**Review by arturcic — CHANGES_REQUESTED** [review]

**Comment by matkoch** [comment]
Updated

File: src/GitVersion.Core.Tests/BuildAgents/SpaceAutomationTests.cs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+        [TearDown]
+        public void TearDown()
+        {
+            environment.SetEnvironmentVariable(SpaceAutomation.EnvironmentVariableName, null);
+            if (setEnvironmentTempFilePath != null && File.Exists(setEnvironmentTempFilePath))
+            {
+                File.Delete(setEnvironmentTempFilePath);
+                setEnvironmentTempFilePath = null;
+            }
```
Comment (arturcic) [inline src/GitVersion.Core.Tests/BuildAgents/SpaceAutomationTests.cs:40]: I think this one is not needed here at all, as only the GH actions build agent is using a file, so it is needed only in the GH tests

File: src/GitVersion.Core.Tests/BuildAgents/SpaceAutomationTests.cs [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        {
+            var sp = ConfigureServices(services =>
+            {
+                services.AddSingleton<SpaceAutomation>();
+            });
+            environment = sp.GetService<IEnvironment>();
+            buildServer = sp.GetService<SpaceAutomation>();
+            environment.SetEnvironmentVariable(SpaceAutomation.EnvironmentVariableName, "true");
+
+            setEnvironmentTempFilePath = Path.GetTempFileName();
```
Comment (arturcic) [inline src/GitVersion.Core.Tests/BuildAgents/SpaceAutomationTests.cs:29]: I think this one is not needed here at all, as only the GH actions build agent is using a file, so it is needed only in the GH tests

File: src/GitVersion.Core.Tests/BuildAgents/SpaceAutomationTests.cs [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+using Shouldly;
+
+namespace GitVersionCore.Tests.BuildAgents
+{
+    [TestFixture]
+    public class SpaceAutomationTests : TestBase
+    {
+        private IEnvironment environment;
+        private SpaceAutomation buildServer;
+        private string setEnvironmentTempFilePath;
```
Comment (arturcic) [inline src/GitVersion.Core.Tests/BuildAgents/SpaceAutomationTests.cs:16]: I think this one is not needed here at all, as only the GH actions build agent is using a file, so it is needed only in the GH tests

**Review by arturcic — CHANGES_REQUESTED** [review]

**Comment by matkoch** [comment]
Updated.

**Review by arturcic — APPROVED** [review]

**Comment by matkoch** [comment]
Just a patch increment?:)

**Pull request merged**